test/geometry_tests.cpp

```diff
@@ -61,105 +94,176 @@ bool checkFirstOrLastPoint(int firstOrLastPoint, int x_shouldbe, int y_shouldbe,
 }
 
 TEST(Point2DPlanning_Griz_Test, bresenham_conversion_point_test) {
 	// Get the points for the line connecting (0,0) and (0,0)
-	auto v = bresenham_conversion(0, 0, 0, 0);
+	const auto v1 = bresenham_conversion(0, 0, 0, 0);
 
 	// Check only one point was produced
-	ASSERT_EQ(v.size(), 1);
+	EXPECT_THAT(v1.size(), testing::Eq(1));
 
 	// Check the point is (0,0)
-	ASSERT_EQ(checkFirstOrLastPoint(0, 0, 0, v), true);	
+	EXPECT_THAT(checkFirstOrLastPoint(0, 0, 0, v1), testing::IsTrue());	
 
 	// Get the points for the line connecting (3,4) and (3,4)
-	v = bresenham_conversion(3, 4, 3, 4);
+	const auto v2 = bresenham_conversion(3, 4, 3, 4);
 
 	// Check only one point was produced
-	ASSERT_EQ(v.size(), 1);
+	EXPECT_THAT(v2.size(), testing::Eq(1));
 
 	// Check the point is (3,4)
-	ASSERT_EQ(checkFirstOrLastPoint(0, 3, 4, v), true);
+	EXPECT_THAT(checkFirstOrLastPoint(0, 3, 4, v2), testing::IsTrue());
 
 }
 
 TEST(Point2DPlanning_Griz_Test, bresenham_conversion_octant1_test) {
 	// Get the points for the line connecting (0,0) and (5,3)
-	auto v = bresenham_conversion(0, 0, 5, 3);
+	const auto v1 = bresenham_conversion(0, 0, 5, 3);
 
 	// Check only one point was produced
-	ASSERT_EQ(v.size(), 6);
+	EXPECT_THAT(v1.size(), testing::Eq(6));
 
 	// Check the first point is (0,0)
-	ASSERT_EQ(checkFirstOrLastPoint(0, 0, 0, v), true);	
+	EXPECT_THAT(checkFirstOrLastPoint(0, 0, 0, v1), testing::IsTrue());	
 
 	// Check the last point is (0,0)
-	ASSERT_EQ(checkFirstOrLastPoint(1, 5, 3, v), true);	
+	EXPECT_THAT(checkFirstOrLastPoint(1, 5, 3, v1), testing::IsTrue());	
 
 	// Get the points for the line connecting (5,3) and (0,0)
-	v = bresenham_conversion(5, 3, 0, 0);
+	const auto v2 = bresenham_conversion(5, 3, 0, 0);
 
 	// Check only one point was produced
-	ASSERT_EQ(v.size(), 6);
+	EXPECT_THAT(v2.size(), testing::Eq(6));
 
 	// Check the first point is (0,0)
-	ASSERT_EQ(checkFirstOrLastPoint(0, 5, 3, v), true);	
+	EXPECT_THAT(checkFirstOrLastPoint(0, 5, 3, v2), testing::IsTrue());	
 
 	// Check the last point is (0,0)
-	ASSERT_EQ(checkFirstOrLastPoint(1, 0, 0, v), true);	
+	EXPECT_THAT(checkFirstOrLastPoint(1, 0, 0, v2), testing::IsTrue());	
+
+}
+
+TEST(Point2DPlanning_Griz_Test, raytraceline_conversion_single_point_test) {
+	const Cell startPix = {0, 0};
+	const Cell endPix = {0, 0};
+	const size_t max_ray_length = 10; ///< This is an arbitrary number
+	const auto pixels = raytraceline_conversion(startPix, endPix, max_ray_length);
+
+	EXPECT_THAT(pixels.size(), testing::Eq(1));	
+}
+
+TEST(Point2DPlanning_Griz_Test, raytraceline_conversion_horz_line_test) {
+	const Cell start_pix_1 = {0, 0};
+	const Cell end_pix_1 = {5, 0};
+	const size_t max_ray_length_1 = 10; ///< This is an arbitrary number
+	const auto pixels_1 = raytraceline_conversion(start_pix_1, end_pix_1, max_ray_length_1);	
+
+	EXPECT_THAT(pixels_1.size(), testing::Eq(6));	
+	EXPECT_THAT(pixels_1,testing::ElementsAre(Cell(0,0),Cell(1,0),Cell(2,0),Cell(3,0),Cell(4,0),Cell(5,0)));
```

Comment:
I'm going to suggest defaulting to using braces for ctors `Cell{0, 0}`, unless you have to use parens (such as certain cases on `std::vector`, (`std::vector<int> v(10, 1)` does something different than `std::vector<int> v{10, 1}`)

Reply:
Otherwise, I fully support using `ElementsAre` as long as it's less than 120 chars on a line (preferably closer to 100 though). See earlier note about `using ::testing::ElementsAre`

Reply:
Done